### PR TITLE
Aop2 57 referral form multi steps

### PR DESCRIPTION
**Pull request by jasonfungsing**

### CODE REVIEW

File: backend/app/api/views/prices.py [hunk 1]
```diff
@@ -195,3 +195,26 @@ def update_ceiling_price(ceiling_id):
...
+# @swag.validate('PriceUpdate')
+def get_price(price_id):
+    price = prices.get(price_id)
+    supplier = suppliers.first(code=price.supplier_code)
+    return jsonify({
+        'supplierCode': price.supplier_code,
+        'supplierName': supplier.name,
+        'categoryName': price.service_type.category.name,
+        'serviceName': price.service_type.name,
+        'regionId': price.region_id,
```
Comment (bsigney) [inline backend/app/api/views/prices.py]: curious why the the `region` wasnt just sent as an object to the ui?

File: frontend/apps/orams/components/ReferralBuilderForms/ReferralBuilder1Form.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+            <Textfield
+              model={`${model}.natureOfInjury`}
+              name="natureOfInjury"
+              id="natureOfInjury"
+              htmlFor="natureOfInjury"
+              label="Nature Of Injury"
+              validators={{ required }}
+              messages={{
+                validABN: 'Nature Of Injury is required'
```
Comment (bsigney) [inline frontend/apps/orams/components/ReferralBuilderForms/ReferralBuilder1Form.js]: why are you using `validABN` validator?  this should be `required` validator

File: frontend/apps/orams/actions/referralActions.js [hunk 3]
```diff
@@ -34,6 +36,9 @@ export const createReferral = data => dispatch => {
       dispatch(setErrorMessage(GENERAL_ERROR))
     } else {
       dispatch(setSuccessMessage(true))
+      dispatch(actions.reset('referralBuilder1Form'))
```
[the remark this answers is not in the document]
Comment (jasonfungsing) [reply frontend/apps/orams/actions/referralActions.js]: yes, the naming in this pr will be all rename after we got the meaningful names from Stephen

File: frontend/apps/orams/components/ReferralBuilderForms/ReferralBuilder1Form.js (hunk 2 above)
Comment (jasonfungsing) [reply frontend/apps/orams/components/ReferralBuilderForms/ReferralBuilder1Form.js]: done

File: frontend/apps/orams/actions/referralActions.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (bsigney) [reply frontend/apps/orams/actions/referralActions.js]: I think the sections already available in his UX screenshot. But yes, check with him as soon as you can

File: frontend/apps/orams/components/ReferralBuilder/ReferralBuilder.js [hunk 4]
```diff
@@ -0,0 +1,237 @@
+/* eslint-disable */
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import PropTypes from 'prop-types'
+import LoadingIndicatorFullPage from 'shared/LoadingIndicatorFullPage/LoadingIndicatorFullPage'
+import styles from './ReferralBuilder.scss'
+import { uniqueID } from 'shared/utils/helpers'
+import AUprogressIndicator from '@gov.au/progress-indicator'
```
Comment (qimyi) [inline frontend/apps/orams/components/ReferralBuilder/ReferralBuilder.js]: Probably don't use AU prefix, I know we have some places already doing that, but moving forward, we can omit it? Thoughts?

File: frontend/apps/orams/reducers/referralDetailsReducer.js [hunk 5]
```diff
@@ -16,6 +22,8 @@ const referralDetailsReducer = (state = initialState, action) => {
   switch (action.type) {
     case SET_REFERRAL_DATA:
       return { ...state, ...action.referralData }
+    case SET_REFERRAL_INFO_DATA:
```
Comment (qimyi) [inline frontend/apps/orams/reducers/referralDetailsReducer.js]: Is this related to price? Maybe rename to something like SET_REFERRAL_PRICING_DATA?

File: frontend/apps/orams/reducers/referralDetailsReducer.test.js [hunk 6]
```diff
@@ -1,12 +1,13 @@
 import { SET_REFERRAL_DATA, SET_LOADING_REFERRAL_DATA, RESET_LOADING_REFERRAL_DATA } from 'orams/constants/constants'
 import referralDetailsReducer from './referralDetailsReducer'
 
-describe.only('referralDetailsReducer', () => {
+describe('referralDetailsReducer', () => {
```
Comment (qimyi) [inline frontend/apps/orams/reducers/referralDetailsReducer.test.js]: add a test case for new action

File: frontend/apps/orams/pages/ReferralBuilderPage.js [hunk 7]
```diff
@@ -48,16 +54,17 @@ ReferralBuilderPage.propTypes = {
...
 
 const mapStateToProps = state => ({
-  supplierData: state.sellersCatalogue.supplierData,
-  regionCode: state.sellersCatalogue.region,
-  regionsData: state.sellersCatalogue.regionsData,
-  price: state.sellersCatalogue.price,
+  referralInfoData: state.referralDetails.referralInfoData,
+  referralBuilder1Form: state.referralBuilder1Form,
+  referralBuilder2Form: state.referralBuilder2Form,
+  referralBuilder3Form: state.referralBuilder3Form,
```
Comment (qimyi) [inline frontend/apps/orams/pages/ReferralBuilderPage.js]: can be simplified to 
```
{referralBuilder1Form, referralBuilder2Form, referralBuilder3Form}: state
```

File: frontend/apps/orams/components/ReferralBuilderForms/ReferralBuilder3Form.js [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+  }
+}
+
+const mapStateToProps = state => {
+  return {
+    ...formProps(state, 'referralBuilder3Form')
+  }
+}
+
+const mapDispatchToProps = dispatch => {
```
Comment (qimyi) [inline frontend/apps/orams/components/ReferralBuilderForms/ReferralBuilder3Form.js]: if returning empty object, this can be omitted

File: frontend/apps/orams/components/ReferralBuilderForms/ReferralBuilder3Form.js [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+  return {}
+}
+
+export { Textfield, mapStateToProps, ReferralBuilder3Form as Form }
+
+export default connect(mapStateToProps, mapDispatchToProps)(ReferralBuilder3Form)
```
Comment (qimyi) [inline frontend/apps/orams/components/ReferralBuilderForms/ReferralBuilder3Form.js]: and omit mapDispatchToProps here

File: frontend/apps/orams/components/ReferralBuilderForms/ReferralBuilder2Form.js [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+  }
+}
+
+const mapStateToProps = state => {
+  return {
+    ...formProps(state, 'referralBuilder2Form')
+  }
+}
+
+const mapDispatchToProps = dispatch => {
```
Comment (qimyi) [inline frontend/apps/orams/components/ReferralBuilderForms/ReferralBuilder2Form.js]: Can be omitted

File: frontend/apps/orams/actions/referralActions.js [hunk 11]
```diff
@@ -1,10 +1,12 @@
+import { actions } from 'react-redux-form'
```
Comment (qimyi) [inline frontend/apps/orams/actions/referralActions.js]: Please add test to referralActions.test.js. Feel free to tap on my shoulder if u need help with that

Comment (jasonfungsing) [reply frontend/apps/orams/actions/referralActions.js]: done

Comment (qimyi) [reply frontend/apps/orams/actions/referralActions.js]: referralActions.test.js test 404 :)

File: backend/app/api/views/prices.py (hunk 1 above)
Comment (waleedusman) [reply backend/app/api/views/prices.py]: This was copied from the branch named `get-price-details` to allow dev to continue. Given that this endpoint is essential for this block of work, I think it needs to be made more production-ready. Would be good to see where else the price needs to be retrieved and to come up with a set of requirements.

File: backend/app/api/views/users.py [hunk 12]
```diff
@@ -57,6 +57,9 @@ def me():
 # deprecated
 @api.route('/ping', methods=["GET"])
 def me_deprecated():
+    if hasattr(current_user, 'email_address'):
```
Comment (waleedusman) [inline backend/app/api/views/users.py]: why are changes being made to a `deprecated` method? which method supersedes this one?

File: backend/tests/fixtures/test_auth.py [hunk 13]
```diff
@@ -2,14 +2,14 @@
...
 
-def test_anonymous(client):
-    res = client.get('/2/ping')
-    data = json.loads(res.get_data(as_text=True))
-
-    assert not data['isAuthenticated']
-
-    res = client.get('/2/_protected')
-    assert res.status_code == 401
+# def test_anonymous(client):
```
Comment (waleedusman) [inline backend/tests/fixtures/test_auth.py]: what are you going to do about this block? uncomment it? delete it?

File: backend/app/api/views/users.py (hunk 12 above)
Comment (jasonfungsing) [reply backend/app/api/views/users.py]: good question, I check around, couldn't see any other endpoint supersede this... and this is ping endpoint has been heavily used by frontend

File: backend/tests/fixtures/test_auth.py (hunk 13 above)
Comment (jasonfungsing) [reply backend/tests/fixtures/test_auth.py]: lo, forgot to uncomment it after my test, on it now

File: backend/app/api/views/users.py (hunk 12 above)
Comment (waleedusman) [reply backend/app/api/views/users.py]: i think the method above supersedes this one i.e. `def me():` or endpoint `GET /users/me`
